Documentation/wip/cli-extensibility.md

```diff
@@ -0,0 +1,128 @@
+3. fission finds the binary.
+4. fission invokes the `fission-workflows`, passing the remainder of the arguments.
+
+**Discoverability: fission --help**
+```bash
+$ fission --help
+USAGE:
+   fission [global options] command [command options] [arguments...]
+
+VERSION:
+   0.6.0
+
+COMMANDS:
+     function, fn                  Create, update and manage functions
+     httptrigger, ht, route        Manage HTTP triggers (routes) for functions
+     timetrigger, tt, timer        Manage Time triggers (timers) for functions
+     mqtrigger, mqt, messagequeue  Manage message queue triggers for functions
+     environment, env              Manage environments
+     watch, w                      Manage watches
+     package, pkg                  Manage packages
+     spec, specs                   Manage a declarative app specification
+     upgrade                       Upgrade tool from fission v0.1
+     tpr2crd                       Migrate tool for TPR to CRD
+     help, h                       Shows a list of commands or help for one command
+
+PLUGINS:
+     workflows, wf                 Inspect and manage workflow executions
+     ui                            Start the user interface
+
+GLOBAL OPTIONS:
+   --server value  Fission server URL (default: "http://127.0.0.1:65356")
+   --help, -h      show help
+   --version, -v   print the version
+```
+Of course Fission needs to be able to find all plugins for this. There are several ways in which we can provide discoverability. The simplest one is for Fission to look in the path for all binaries starting with the `fission-*` prefix. Optionally, fission could invoke a specific command on the subcommand to get info about the plugin (such as version, help text, aliases...)
+
+With Fission Workflows this info would look something like this: 
+```bash
+$ fission-workflows --plugin
+name: workflows
+version: 0.4.0
+help: Inspect and manage workflow executions
+```
+The idea is that this plugin info is all completely optional. 
+If it is not available, we simply degrade the results to user. 
+This way users/we can easily prototype or add plugins without having to worry about adhering to some interface.
+
+**List version**
+```bash
+$ fission --version
+client: 
+  fission: 0.8.0
+  fission-workflows: 0.4.0
+server:
+  fission: 0.8.1
+  fission-workflows: 0.3.0
+```
+Again, versioning info for fission-workflows is taken from the plugin info of the commands. 
+Note: a related issue is to have some more formalized plugin support/discoverability on the server-side, 
+but that is out of the scope of this issue.
+
+### Other (optional) extensions and notes
+- Like git we could setup a preferred binary path, where fission looks first when searching for the subcommand. 
+This could optionally be defined with a `FISSION_EXEC_PATH`.
+- With the current approach we cannot have aliases for commands---fission will not be able to find fission-workflows 
```

Comment:
the --plugin output can contain aliases

Reply:
Yes, this document was just a direct copy-paste of the original proposal in issue #743. This part is outdated, plugin metadata can indeed contain aliases. 

I'll update this document before merging.